src/package_json.rs

```diff
@@ -125,85 +64,125 @@ impl PackageJson {
-                // Using `get` + `clone` instead of remove here
-                // because `main_fields` may contain `browser`, which is also used in `browser_fields.
-                if let Some(serde_json::Value::String(value)) =
-                    package_json_value.get(main_field_key)
-                {
-                    package_json.main_fields.push(value.clone());
+        // Dynamically create `main_fields`.
+        for main_field_key in &options.main_fields {
+            match main_field_key.as_str() {
+                "main" => {
+                    if let Some(main_field) = &data.main {
+                        package_json.main_fields.push(main_field.to_string_lossy().to_string());
+                    }
                 }
-            }
+                "module" => {
+                    if let Some(module_field) = &data.module {
+                        package_json.main_fields.push(module_field.to_string_lossy().to_string());
+                    }
+                }
+                "browser" => {
+                    if let Some(BrowserField::String(browser_field)) = &data.browser {
+                        package_json.main_fields.push(browser_field.to_owned());
+                    }
+                }
+                field => {
+                    // Using `get` + `clone` instead of remove here
+                    // because `main_fields` may contain `browser`, which is also used in `browser_fields.
+                    if let Some(Value::String(value)) = data.other_fields.get(field) {
+                        package_json.main_fields.push(value.clone());
+                    }
+                }
+            };
         }
 
         // Dynamically create `browser_fields`.
         let dir = path.parent().unwrap();
         for object_path in &options.alias_fields {
-            if let Some(browser_field) = Self::get_value_by_path(&package_json_value, object_path) {
-                let mut browser_field = BrowserField::deserialize(browser_field)?;
-                // Normalize all relative paths to make browser_field a constant value lookup
-                if let BrowserField::Map(map) = &mut browser_field {
-                    let keys = map.keys().cloned().collect::<Vec<_>>();
-                    for key in keys {
-                        // Normalize the key if it looks like a file "foo.js"
-                        if key.extension().is_some() {
-                            map.insert(dir.normalize_with(&key), map[&key].clone());
-                        }
-                        // Normalize the key if it is relative path "./relative"
-                        if key.starts_with(".") {
-                            if let Some(value) = map.remove(&key) {
-                                map.insert(dir.normalize_with(&key), value);
-                            }
+            let mut browser_field = if object_path.len() == 1 && object_path[0] == "browser" {
```

Comment:
Uses the browser field directly: https://docs.rs/nodejs_package_json/0.1.1/nodejs_package_json/struct.PackageJson.html#structfield.browser